File: stocks-api/src/main/kotlin/com/cleverskills/domain/house/House.kt
package com.cleverskills.domain.house

import java.time.LocalDate

data class House(
    val id: Long,
    val type: HouseType,
    val name: String,
    val acquisitionPrice: Long,
    val acquisitionDate: LocalDate,
    val city: String,
    val userId: String,
    val projectId: Long?,
)

File: stocks-frontend/src/app/domain/model/House.ts
export interface House {
  acquisitionDate: string;
  id: string | null;
  type: HouseType;
  name: string;
  acquisitionPrice: number;
  city: string;
  userId: string;
  projectId: string | null;
}

export enum HouseType {
  APPART,
  MAISON,
  IMMEUBLE,
  BATEAU,
}


export function getHouseTypeKeys() {
  return Object.keys(HouseType).filter(f => {
    return !isNaN(Number(f))
  }).map(value => {
    return parseInt(value)
  })
}

export function getHouseTypeValues() {
  return Object.keys(HouseType).filter(f => {
    return isNaN(Number(f))
  })
}
